addons/phone_validation/models/phone_blacklist.py

```diff
@@ -26,29 +28,46 @@ class PhoneBlackList(models.Model):
 
     @api.model_create_multi
     def create(self, values):
-        # First of all, extract values to ensure emails are really unique (and don't modify values in place)
-        to_create = []
-        done = set()
+        """ Create new (or activate existing) blacklisted numbers.
+                A. Note: Attempt to create a number that already exists, but is non-active, will result in its activation.
+                B. Note: If the number already exists and is active it (won't be created) will be added to returned set
+        Returns Recordset of: created, activated, existing phonenumbers from the requested list of numbers to create
+        """
+        # Extract and sanitize numbers, ensuring uniques
+        to_create = set()
         for value in values:
             number = value['number']
             sanitized_values = phone_validation.phone_sanitize_numbers_w_record([number], self.env.user)[number]
             sanitized = sanitized_values['sanitized']
             if not sanitized:
                 raise UserError(sanitized_values['msg'] + _(" Please correct the number and try again."))
-            if sanitized in done:
+            if sanitized in to_create:
                 continue
-            done.add(sanitized)
-            to_create.append(dict(value, number=sanitized))
```

Comment:
since there is only two field, the only intent of this line seems to keep value of the second field (`active`) and override `number`

the current version doesn't take `active` into account, and would not do what is intended for the line:

https://github.com/odoo/odoo/blob/82b03c37327abbab0b0bb63ba2423fe475004b66/addons/phone_validation/models/phone_blacklist.py#L117

Reply:
Good point, thanks for noticing!

I will preserve original construct (keeping `to_create` as dict). I see how this can ease future changes or customization